docs/paypal.md

```diff
@@ -0,0 +1,173 @@
+
+### Technical steps - custom integration
+
+#### 1. Add the Payments SDK  to your app
+
+#### Swift Package Manager
+
+To add the PayPal package to your Xcode project, select _File > Swift Packages > Add Package Dependency_ and enter https://github.com/paypal/iOS-SDK as the repository URL. Tick the checkboxes for the specific PayPal libraries you wish to include.
+
+In your app's source code files, use the following import syntax to include PayPal's libraries:
+
+```swift
+import PayPal
+```
+
+#### CocoaPods
+
+Include the PayPal pod in your `Podfile`.
+
+```ruby
+pod 'PayPal'
+```
+
+In your app's source code files, use the following import syntax to include PayPal's libraries:
+
+```swift
+import PayPal
+```
+
+#### 2. Create a PayPal button 
+
+Add a `PayPalButton` to your View:
+
+```swift
+let payPalButton = PayPalButton()
+```
+
+#### 3. Initiate the Payments SDK
+
+Create a `CoreConfig` using your client ID from the PayPal Developer Portal:
+
+```swift
+let config = CoreConfig(clientID: "<CLIENT_ID>", environment: .sandbox)
+```
+
+Configure a return URL using your bundle identifier:
+
+```swift
+let returnUrl = "<BUNDLE_IDENTIFIER>" + "://paypalpay"
+```
+
+Create a `PayPalClient` to approve an order with a PayPal payment method:
+
+```swift
+let payPalClient = PayPalClient(config: config, returnURL: returnUrl)
+```
+
+#### 4. Create an order
+
+When a user enters the payment flow, call `v2/checkout/orders` to create an order and obtain an order ID:
+
+```bash
+curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/orders/' \
+--header 'Content-Type: application/json' \
+--header 'Authorization: Basic <ENCODED_CLIENT_ID>' \
+--data-raw '{
+    "intent": "CAPTURE",
+    "purchase_units": [
+        {
+            "amount": {
+                "currency_code": "USD",
+                "value": "5.00"
+            }
+        }
+    ]
+}'
+```
+
+The `id` field of the response contains the order ID to pass to your client.
+
+#### 5. Approve the order through the Payments SDK
+
+When a user taps the PayPal button created above, approve the order using your `PayPalClient`.
+
+Add a target to your PayPal button to launch the PayPal payment flow:
+
+```swift
+payPalButton.addTarget(self, action: #selector(paymentButtonTapped), for: .touchUpInside)
+
+@objc func paymentButtonTapped() {
+    checkoutWithPayPal(orderID: "<ORDER_ID>")
+}
+```
+
+Call `PayPalClient.start` to approve the order, and then handle results:
+
+```swift
+func checkoutWithPayPal(orderID: String) {
+    payPalClient.start(orderID: orderID, presentingViewController: self) { state in
+        switch state {
+        case .success(let result):
+            // capture/authorize the order using `result.orderID` (see step 6)        
+        case .failure(let error):
+            // handle the error by accessing `error.localizedDescription`
+        case .cancellation:
+            // the user canceled
```

Comment:
Need to confirm if we can move forward with adding cancellation to the failure case. This had been discussed previously.